vllm/engine/arg_utils.py

```diff
@@ -65,6 +65,23 @@ def _optional_type(val: str) -> Optional[T]:
     return _optional_type
 
 
+def type_or_type_list(val: str, return_type: Callable[[str],
+                                                      T]) -> Union[T, List[T]]:
+    try:
+        # First try parsing as comma-separated list
+        if ',' in val:
+            return [return_type(x) for x in val.split(',')]
+        # Otherwise, treat as single int
+        return return_type(val)
+    except ValueError as e:
+        raise argparse.ArgumentTypeError(
+            "Must be a type or comma-separated list of types") from e
+
+
+def int_or_int_list(val: str) -> Union[int, List[int]]:
+    return type_or_type_list(val, int)
```

Comment:
I'm not sure if these are necessary:
- Comma separated lists are not very Pythonic when we have `nargs="+"`
- If the type hint is a list `get_kwargs` will automatically set `nargs="+"`